File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        if(cacheSize == 0){
            return cities.length * 5;
        }
        
        List<String> list = new ArrayList<>();
        
        for(int i=0; i<cities.length; i++){
            if(list.size() < cacheSize && !list.contains(cities[i].toLowerCase())){
                list.add(cities[i].toLowerCase());
                answer += 5;
            }
            else if(list.size() < cacheSize && list.contains(cities[i].toLowerCase())){
                list.remove(cities[i].toLowerCase());
                list.add(cities[i].toLowerCase());
                answer += 1;
            }
            else if(list.size() >= cacheSize && !list.contains(cities[i].toLowerCase())){
                list.remove(0);
                list.add(cities[i].toLowerCase());
                answer += 5;
            }
            else if(list.size() >= cacheSize && list.contains(cities[i].toLowerCase())){
                list.remove(cities[i].toLowerCase());
                list.add(cities[i].toLowerCase());
                answer += 1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    var answer = 0;
    let queue = [];
    if(cacheSize == 0){
        return cities.length * 5;
    }
    for(let i of cities){
        if(queue.includes(i.toUpperCase())){
            queue.splice(queue.indexOf(i.toUpperCase()),1);
            queue.push(i.toUpperCase());
            answer+=1;
        }
        else{
            if(queue.length == cacheSize){
                queue.shift();
            }
            queue.push(i.toUpperCase());
            answer+=5;
        }
        
    }
    return answer;
}